File: src/ScheduleTime.Domain/Entities/Customer.cs
namespace ScheduleTime.Domain.Entities
{
    public class Customer
    {
        public long Id { get; private set; }
        public string FirstName { get; private set; } = string.Empty;
        public string LastName { get; private set; } = string.Empty;
        public string FullName { get; private set; } = string.Empty;
        public string Email { get; private set; } = string.Empty;
        public string Phone { get; private set; } = string.Empty;

        private Customer() { }

        public Customer(string firstName, string lastName, string fullName, string email, string phone)
        {
            FirstName = firstName;
            LastName = lastName;
            FullName = fullName;
            Email = email;
            Phone = phone;
        }

        public static class Factories
        {
            public static Customer CreateCustomer(Customer customer)
            {
                return new Customer(customer.FirstName, customer.LastName,
                    customer.FullName, customer.Email, customer.Phone);
            }
        }
    }
}

File: java/schedule-time-event/src/main/java/com/gearguru/schedule_time_event/Domain/Entities/Customer.java
package com.gearguru.schedule_time_event.Domain.Entities;

import lombok.*;

@Getter(AccessLevel.PUBLIC)
@Setter(AccessLevel.PRIVATE)
@NoArgsConstructor
public class Customer {
    private Long id;
    private final String firstName;
    private final String lastName;
    private final String fullName;
    private final String email;
    private final String phone;

    public Customer(String firstName, String lastName, String fullName, String email, String phone) {
        this.firstName = firstName;
        this.lastName = lastName;
        this.fullName = fullName;
        this.email = email;
        this.phone = phone;
    }
}
